### ISSUE TITLE
[Bug] Non-optimal regular expression: Suggested fix "strpos can be used instead" is wrong

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1064]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug                          |
| Plugin         | Php Inspections (EA Extended), 3.0.6.1 |
| Language level | PHP 7.2                        |

Current behaviour:
---

Here is some example code:

```php
function foo(string $input): string
{
    if (preg_match('/test/', $input) === 1) {
        return 'Input includes "test".';
    }
    return 'Input does not include "test".';
}
```

When using the suggested fix from the "Non-optimal regular expression" inspection ("'false !== strpos($input, "test")' can be used instead.") then the code is changed into that:

```php
function foo(string $input): string
{
    if (false !== strpos($input, "test") === 1) {
        return 'Input includes "test".';
    }
    return 'Input does not include "test".';
}
```

That applied suggestion looks pretty strange to me. It's also broken: See https://3v4l.org/RPh5p

Expected behaviour:
---

I'd actually expect the suggestion to change the code into this:

```php
function foo(string $input): string
{
    if (strpos($input, 'test') !== false) {
        return 'Input includes "test".';
    }
    return 'Input does not include "test".';
}
```

1.) The bug: The `=== 1` part has to be removed. Because after applying the suggestion, the code is obviously broken.
2.) Why does the suggestion use double-quoted instead of single-quotes here? (Okay, not a big deal, but still.)
3.) I set the yoda switch setting correctly in my settings, still the suggestion applied yoda style.

**Comment by kalessil** [comment]
Confirmed

**Comment by kalessil** [comment]
Working on it, good catch Andreas!

**Comment by kalessil** [comment]
Fixed!

**Comment by andreasschroth** [comment]
Great, thank you! 👍
